plugin.php

```diff
@@ -584,12 +588,24 @@ function rest_handle_options_request( $response, $handler, $request ) {
 
 		foreach ( $endpoints as $endpoint ) {
 			$accept = array_merge( $accept, $endpoint['methods'] );
+
+			$methods = array_keys( $endpoint['methods'] );
+			$method = array_shift( $methods );
+
+			if ( empty( $body['request_args'][ $method ] ) ) {
+				$body['request_args'][ $method ] = $endpoint['args'];
```

Comment:
Just mentioned it in my comment too! I think yes, a map with `WP_REST_Server::CREATABLE` and all those constants would be best I think.